### PR TITLE
Adapt data problem template to new form style

### PR DESCRIPTION
**Pull request by queengooborg**
Inspired by https://github.com/mdn/content/blob/main/.github/ISSUE_TEMPLATE/content-bug.yml, this PR adapts our issue template to the new issues form style.

### CODE REVIEW

**Comment by foolip** [comment]
I think this will be a nice improvement! https://github.com/mdn/browser-compat-data/blob/main/.github/ISSUE_TEMPLATE/data-problem.md is the current template, and turning that into a form would be nice.

It's good that it's still possible to open a blank issue, which is often what I want when things get complicated.

File: .github/ISSUE_TEMPLATE/data-problem.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        **Need help with a browser?**
+        🙋 To get help with [Firefox](https://support.mozilla.org/en-US/kb/file-bug-report-or-feature-request-mozilla), [Chrome](https://support.google.com/chrome/answer/95315?hl=en-GB&ref_topic=7439544), [Safari](https://www.apple.com/feedback/safari.html), or another browser, check the browser's support site.
+
+        ---
+  - type: dropdown
+    id: type
+    attributes:
+      label: What type of issue is this?
+      options:
+        - label: Incorrect version number (ex. Chrome says "42" but support was added in "40")
```
Comment (foolip) [inline .github/ISSUE_TEMPLATE/data-problem.yml:26]: Maybe update the versions here to be more contemporary, and also a bigger error than 2 versions for more realism?

File: .github/ISSUE_TEMPLATE/data-problem.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  - type: checkboxes
+    id: browsers
+    attributes:
+      label: What browsers does this problem apply to, if applicable?
+      multiple: true
+      options:
+        - Chromium (Chrome, Edge, Opera, Samsung Internet)
+        - Firefox
+        - Safari
+        - NodeJS
```
Comment (foolip) [inline .github/ISSUE_TEMPLATE/data-problem.yml:44]:
```suggestion
        - Node.js
```

File: .github/ISSUE_TEMPLATE/data-problem.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        - Deno
+  - type: textarea
+    id: expected
+    attributes:
+      label: What did you expect to see?
+    validations:
+      required: true
```
Comment (foolip) [inline .github/ISSUE_TEMPLATE/data-problem.yml:51]: I don't think this should be required. It's often implied by saying what is wrong, and having to repeat is annoying. Or maybe I just know that it's wrong, but not what is correct.

**Review by Rumyra — APPROVED** [review]

**Comment by queengooborg** [comment]
I'm going to go ahead and merge this since there is owner approval (and I'm getting sick of closing more spam pull requests).  I'm happy to make any follow-up changes as needed!

**Pull request merged**